utils.php

```diff
@@ -24,4 +24,33 @@ function i8ln($word)
     } else {
         return $word;
     }
+}
+
+function getAndSetToken()
+{
+    if (! isset($_SESSION['token'])) {
+        generateToken();
+    }
+}
```

Comment:
`getAndSetToken` is a dubious name since it doesn't return anything.

For clarity and readability, you can change it to `setSessionCsrfToken($token)` and use dependency injection. I use `empty(...)` because it return `true` for any variable that isn't set or that is empty:
```
if (empty($_SESSION['token'])) {
    setSessionCsrfToken(generateCsrfToken());
}
```
In this example, I also renamed `generateToken` to `generateCsrfToken` so the function is as clear as it can be. Same goes for `validateToken`.